### DISCUSSION TITLE
Struggling a bit with how to handle route enum

### DISCUSSION THREAD

**Discussion by d-pollard** [remix-run/react-router#8826] [v5 to v6 Migration]
I seem to be struggling a bit with how I should convert my route paths enum to be better utilized in v6

v5 route paths:

```typescript
export enum RoutePaths {
  Workspaces = '/workspaces/:workspaceId?',
}
```

So taking `Workspaces` as the example issue, in v5, it would cover cases `/workspaces` and `/workspaces/1234` however, in v6, this only covers `/workspaces/1234`

Does this mean I need to add _another_ entry to my enum that's just `WorkspacesBase` that then contains just `/workspaces` and then in addition, when creating the route, I need to use `<Route path={RoutePaths.WorkspacesBase + '*'} />` ?

Is there maybe a more concise way of handling this so I don't need to maintain multiple entries of `workspaces` route?

Any thoughts are greatly appreciated!

**Comment by TheWashiba** [comment]
Yes, that's the current way of handling route enums actually. Since the removal of regex based matching, it's just how you need to write paths on different depth levels. Those who used full path enums in diff depth levels of route components with v5 now also have to manually split URL strings by segments and place accordingly.
